src/Task1/SelectGenre.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import "./selectGenre.css";
+const SelectGenre = () => {
+  const [show, setShow] = useState("");
```

Comment:
Seems like this state is actually boolean value. From what I see, this state is needed to track whether drop-down is visible or not. So simply applying `isDropdownVisible` or even `isOpen` boolean state would be more appropriate. Then, to mark it visible, you will just call it `setIsOpen(true)`, and `false` would be for hidden. Just to toggle its value, you can use `setIsOpen(previousIsOpen => !previousIsOpen)`

Reply:
It is not what you think. This state takes two value one "" second "visible" that is class name, according to class name I have style.

Reply:
Well, I got that, but that doesn't change anything in what I said. The essence of your state here is openness state. Visibility class-name for custom styling is actually a derived value from this core state. In fact, you also have another derived value from this same state custom styling of an arrow icon - custom class-name for drop-down arrow icon (up and down).

https://brewhouse.io/blog/2015/03/24/best-practices-for-component-state-in-reactjs.html - this article may give you some more info on this. The examples are based on class components, but the idea stays the same. So I think you can easily think of how it corresponds to functional components. You can read only the 2nd part of this article - "Store the simplest possible values to describe a component’s state", that's the part relevant to our discussion. However, if you have time, I would suggest you reading the whole article. It includes some other useful information too.

Reply:
I did as you said.